rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py

```diff
@@ -562,6 +562,59 @@ def complete_robot_action(self):
             self.node.get_logger().info(f"Robot {self.name} has completed the"
                                         f" action it was performing")
 
+    def newly_closed_lanes(self, closed_lanes):
+        need_to_replan = False
+        current_lane = self.get_current_lane()
+
+        if self.target_waypoint is not None and \
+                self.target_waypoint.approach_lanes is not None:
+            for lane_idx in self.target_waypoint.approach_lanes:
+                if lane_idx in closed_lanes:
+                    need_to_replan = True
+                    # The robot is currently on a lane that has been closed.
+                    # We take this to mean that the robot needs to reverse.
+                    if lane_idx == current_lane:
+                        lane = self.graph.get_lane(current_lane)
+                        wp0 = self.graph.get_waypoint(
+                            lane.entry.waypoint_index).location
+                        wp1 = self.graph.get_waypoint(
+                            lane.exit.waypoint_index).location
+
+                        position = self.position
+                        return_waypoint = wp0.index
+                        reverse_lane = \
+                            self.graph.lane_from(wp1.index, wp0.index)
+
+                        if reverse_lane:
+                            # We know what lane will reverse us back to the
+                            # beginning of our current lane, so we will update
+                            # our position by saying that we are on that lane.
+                            lanes = []
+                            lanes.append(reverse_lane.index)
+                            self.update_handle.update_current_lanes(
```

Comment:
Wouldn't this only update the lane state one time while the main thread will continue calling update_current_lanes() with `self.on_lane` ? So we should technically just set `self.on_lane = reverse_lane.index`. Similarly in the `else` we would just need to update the `self.target_waypoint = return_waypoint` ? The main update loop will do the updating.

The way we track the robot in this adapter is different from full_control

Reply:
Ah thanks for pointing that out, I see what you mean. Updated accordingly!